exporters/otlp/otlplog/otlploggrpc/internal/selfobservability/selfobservability.go

```diff
@@ -0,0 +1,106 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package selfobservability provides self-observability metrics for OTLP log exporters.
+// This is an experimental feature controlled by the x.SelfObservability feature flag.
+package selfobservability // import "go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc/internal/selfobservability"
+
+import (
+	"context"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/codes"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk"
+	semconv "go.opentelemetry.io/otel/semconv/v1.36.0"
+	"go.opentelemetry.io/otel/semconv/v1.36.0/otelconv"
+)
+
+type ExporterMetrics struct {
+	logInflightMetric         otelconv.SDKExporterLogInflight
+	logExportedMetric         otelconv.SDKExporterLogExported
+	logExportedDurationMetric otelconv.SDKExporterOperationDuration
+	presetAttrs               []attribute.KeyValue
+}
+
+func NewExporterMetrics(
+	name, componentName string,
+	componentType otelconv.ComponentTypeAttr,
+	target string,
+) *ExporterMetrics {
+	em := &ExporterMetrics{}
+	mp := otel.GetMeterProvider()
+	m := mp.Meter(
+		name,
+		metric.WithInstrumentationVersion(sdk.Version()),
+		metric.WithSchemaURL(semconv.SchemaURL))
+
+	var err error
+	if em.logInflightMetric, err = otelconv.NewSDKExporterLogInflight(m); err != nil {
+		otel.Handle(err)
+	}
+	if em.logExportedMetric, err = otelconv.NewSDKExporterLogExported(m); err != nil {
+		otel.Handle(err)
+	}
+	if em.logExportedDurationMetric, err = otelconv.NewSDKExporterOperationDuration(m); err != nil {
+		otel.Handle(err)
+	}
+
+	em.presetAttrs = []attribute.KeyValue{
+		semconv.OTelComponentName(componentName),
+		semconv.OTelComponentTypeKey.String(string(componentType)),
+	}
+	em.presetAttrs = append(em.presetAttrs, ServerAddrAttrs(target)...)
+	return em
+}
+
+func (em *ExporterMetrics) TrackExport(ctx context.Context, count int64) func(err error, code codes.Code) {
+	attrs := append([]attribute.KeyValue{}, em.presetAttrs...)
+
+	begin := time.Now()
+	em.logInflightMetric.Add(ctx, count, attrs...)
+	return func(err error, code codes.Code) {
+		duration := time.Since(begin).Seconds()
+		em.logInflightMetric.Add(ctx, -count, attrs...)
+		if err != nil {
+			attrs = append(attrs, semconv.ErrorType(err))
+		}
+		em.logExportedMetric.Add(ctx, count, attrs...)
+		attrs = append(
+			attrs,
+			em.logExportedDurationMetric.AttrRPCGRPCStatusCode(otelconv.RPCGRPCStatusCodeAttr(code)),
+		)
+		em.logExportedDurationMetric.Record(ctx, duration, attrs...)
+	}
+}
+
+func ServerAddrAttrs(target string) []attribute.KeyValue {
```

Comment:
`target` [always has the form `"<scheme>://[authority]/<endpoint>"`](https://pkg.go.dev/google.golang.org/grpc#ClientConn.CanonicalTarget). Why not just use `url.Parse` to handle all schema values?